File: tests/rosetta/transpiler/php/checkpoint-synchronization-3.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function lower($ch) {
  global $partList, $nAssemblies, $a;
  $up = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
  $low = 'abcdefghijklmnopqrstuvwxyz';
  $i = 0;
  while ($i < strlen($up)) {
  if ($ch == substr($up, $i, $i + 1 - $i)) {
  return substr($low, $i, $i + 1 - $i);
}
  $i = $i + 1;
};
  return $ch;
};
  $partList = ['A', 'B', 'C', 'D'];
  $nAssemblies = 3;
  foreach ($partList as $p) {
  echo rtrim($p . ' worker running'), PHP_EOL;
}
  for ($cycle = 1; $cycle < ($nAssemblies + 1); $cycle++) {
  echo rtrim('begin assembly cycle ' . _str($cycle)), PHP_EOL;
  $a = '';
  foreach ($partList as $p) {
  echo rtrim($p . ' worker begins part'), PHP_EOL;
  echo rtrim($p . ' worker completed ' . strtolower($p)), PHP_EOL;
  $a = $a . strtolower($p);
};
  echo rtrim($a . ' assembled.  cycle ' . _str($cycle) . ' complete'), PHP_EOL;
}
  foreach ($partList as $p) {
  echo rtrim($p . ' worker stopped'), PHP_EOL;
}
$__end = _now();
$__end_mem = memory_get_usage();
$__duration = intdiv($__end - $__start, 1000);
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;;

File: tests/rosetta/transpiler/Clojure/checkpoint-synchronization-3.clj
(ns main (:refer-clojure :exclude [lower]))

(require 'clojure.set)

(defn in [x coll]
  (cond (string? coll) (clojure.string/includes? coll x) (map? coll) (contains? coll x) (sequential? coll) (some (fn [e] (= e x)) coll) :else false))

(defn padStart [s w p]
  (loop [out (str s)] (if (< (count out) w) (recur (str p out)) out)))

(def nowSeed (atom (let [s (System/getenv "MOCHI_NOW_SEED")] (if (and s (not (= s ""))) (Integer/parseInt s) 0))))

(declare lower)

(declare a lower_i lower_low lower_up nAssemblies partList)

(defn lower [lower_ch]
  (try (do (def lower_up "ABCDEFGHIJKLMNOPQRSTUVWXYZ") (def lower_low "abcdefghijklmnopqrstuvwxyz") (def lower_i 0) (while (< lower_i (count lower_up)) (do (when (= lower_ch (subs lower_up lower_i (+ lower_i 1))) (throw (ex-info "return" {:v (subs lower_low lower_i (+ lower_i 1))}))) (def lower_i (+ lower_i 1)))) (throw (ex-info "return" {:v lower_ch}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(def partList ["A" "B" "C" "D"])

(def nAssemblies 3)

(defn -main []
  (doseq [p partList] (println (str p " worker running")))
  (doseq [cycle (range 1 (+ nAssemblies 1))] (do (println (str "begin assembly cycle " (str cycle))) (def a "") (doseq [p partList] (do (println (str p " worker begins part")) (println (str (str p " worker completed ") (lower p))) (def a (str a (lower p))))) (println (str (str (str a " assembled.  cycle ") (str cycle)) " complete"))))
  (doseq [p partList] (println (str p " worker stopped"))))

(-main)
